File: Cliente1/CreditoManagement.vb
Imports AccesoDatos
Imports Entidades
Public Class CreditoManagement
    Public crudCredito As AccesoDatos.Crud.CreditoCrudFactory
    Public Sub CreditoManagement()
        crudCredito = New AccesoDatos.Crud.CreditoCrudFactory()
    End Sub

    Public Function Create(credito As Entidades.Credito)
        crudCredito.Create(credito)
    End Function

    Public Function RetrieveAll() As List(Of Entidades.Credito)
        Return crudCredito.RetrieveAll(Of Entidades.Credito)
    End Function

    Public Function RetrieveById(credito As Entidades.Credito) As Entidades.Credito
        Return crudCredito.Retrieve(Of Entidades.Credito)
    End Function

    Friend Function Update(credito As Entidades.Credito)
        crudCredito.Update(credito)
    End Function

    Friend Function Delete(credito As Entidades.Credito)
        crudCredito.Delete(credito)
    End Function

End Class

File: Management/CreditoManagement.cs
using AccesoDatos.Crud;
using Entidades;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Management
{
    public class CreditoManagement
    {
        private CreditoCrudFactory crudCredito;

        public CreditoManagement()
        {
            crudCredito = new CreditoCrudFactory();
        }

        public void Create(Credito credito)
        {

            crudCredito.Create(credito);

        }


        public List<Credito> RetrieveAll()
        {
            return crudCredito.RetrieveAll<Credito>();
        }

        public Credito RetrieveById(Credito credito)
        {
            return crudCredito.Retrieve<Credito>(credito);
        }

        public void Update(Credito credito)
        {
            crudCredito.Update(credito);
        }

        public void Delete(Credito credito)
        {
            crudCredito.Delete(credito);
        }
    }
}
